docs/technical_architecture.md

```diff
@@ -0,0 +1,68 @@
+# Technical architecture
+
+## High level overview
+
+The scraper behavior is done in two phases.
+
+First the WARC records are iterated to compute the ZIM metadata (find main path, favicon, ...)
+and detect which ZIM paths are expected to be populated (mandatory to know when we will rewrite the documents if the
+URL has been scrapped or not, since it might be customized by the end-user).
```

Comment:
I think it should be reworded (customized by end-user is a detail already assuming doc reader expects domain specific).
We need to know so we only rewrite what leads into the ZIM so external links continue to be external)